interfaces/HO-Portal/src/app/program/make-payment/make-payment.component.ts

```diff
@@ -75,28 +76,21 @@ export class MakePaymentComponent implements OnInit {
   private checkIsEnabled(): boolean {
     this.isEnabled =
       this.totalIncluded > 0 &&
-      this.lastPaymentId < this.program.distributionDuration &&
+      this.payment <= this.program.distributionDuration &&
       !this.paymentInProgress;
     return this.isEnabled;
   }
 
-  private async getNextPaymentId(): Promise<number> {
-    let previousId = 0;
-
-    if (this.lastPaymentId > 0) {
-      previousId = this.lastPaymentId;
-    }
-
-    return previousId + 1;
-  }
-
   public async performPayment(): Promise<void> {
     this.isInProgress = true;
 
-    const nextPaymentId = await this.getNextPaymentId();
+    const paymentId =
+      this.payment ||
+      (await this.pastPaymentsService.getNextPaymentId(this.program));
+    const referenceIds = this.referenceIds.length ? this.referenceIds : null;
```

Comment:
This can be handled inside `this.programsService.submitPayout()` I think, as it has to be done only once for all use-cases.